app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -554,41 +556,112 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
-                commitChange(resources.getString(R.string.git_commit_remove_text, item.longName))
-                deletePasswords(selectedItems)
-            }
-            .setNegativeButton(resources.getString(R.string.dialog_no)) { _, _ ->
-                deletePasswords(selectedItems)
+                commitChange(resources.getString(R.string.git_commit_remove_text,
+                    selectedItems.joinToString(separator = ", ") { item ->
+                        item.file.toRelativeString(getRepositoryDirectory(this))
+                    }
+                ))
+                refreshPasswordList()
             }
+            .setNegativeButton(resources.getString(R.string.dialog_no), null)
             .show()
     }
 
     fun movePasswords(values: List<PasswordItem>) {
         val intent = Intent(this, SelectFolderActivity::class.java)
-        val fileLocations = ArrayList<String>()
-        for ((_, _, _, file) in values) {
-            fileLocations.add(file.absolutePath)
-        }
+        val fileLocations = values.map { it.file.absolutePath }.toTypedArray()
         intent.putExtra("Files", fileLocations)
         intent.putExtra(BaseGitActivity.REQUEST_ARG_OP, "SELECTFOLDER")
-        startActivityForResult(intent, REQUEST_CODE_SELECT_FOLDER)
+        registerForActivityResult(StartActivityForResult()) { result ->
+            val intentData = result.data ?: return@registerForActivityResult
+            val filesToMove = requireNotNull(intentData.getStringArrayExtra("Files"))
+            val target = File(requireNotNull(intentData.getStringExtra("SELECTED_FOLDER_PATH")))
+            val repositoryPath = getRepositoryDirectory(applicationContext).absolutePath
+            if (!target.isDirectory) {
+                e { "Tried moving passwords to a non-existing folder." }
+                return@registerForActivityResult
+            }
+
+            d { "Moving passwords to ${intentData.getStringExtra("SELECTED_FOLDER_PATH")}" }
+            d { filesToMove.joinToString(", ") }
+
+            lifecycleScope.launch(Dispatchers.IO) {
+                for (file in filesToMove) {
+                    val source = File(file)
+                    if (!source.exists()) {
+                        e { "Tried moving something that appears non-existent." }
+                        continue
+                    }
+                    val destinationFile = File(target.absolutePath + "/" + source.name)
+                    val basename = source.nameWithoutExtension
+                    val sourceLongName = getLongName(requireNotNull(source.parent), repositoryPath, basename)
+                    val destinationLongName = getLongName(target.absolutePath, repositoryPath, basename)
+                    if (destinationFile.exists()) {
+                        e { "Trying to move a file that already exists." }
+                        withContext(Dispatchers.Main) {
+                            MaterialAlertDialogBuilder(this@PasswordStore)
```

Comment:
Aren't we going to show multiple dialogs concurrently here? If that works and they stack correctly, that would probably be fine, otherwise we should use `suspendCoroutine` here.

Reply:
Oh yeah, totally missed that. Will confirm that in a bit.

Reply:
Just tested it, the dialogs stack correctly and respect the dialog choices as well.